src/llvm-final-gc-lowering.cpp

```diff
@@ -208,6 +208,7 @@ Value *FinalLowerGC::lowerGCAllocBytes(CallInst *target, Function &F)
         newI = builder.CreateCall(poolAllocFunc, { ptls, pool_offs, pool_osize });
     }
     newI->setAttributes(newI->getCalledFunction()->getAttributes());
+    newI->addRetAttr(Attribute::getWithDereferenceableBytes(F.getContext(),offset));
```

Comment:
Does this actually do much? Very few optimizations run after this point (no harm in this of course).

Reply:
I wonder if that attribute could be added earlier? I will take a look